ochazuke/webhooks/helpers.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""Helpers methods for webhooks."""
+
+import hashlib
+import hmac
+import logging
+
+from datetime import datetime
+
+from ochazuke import app
+from ochazuke.models import db, Milestone, Label, Issue, Event
+
+
+def get_payload_signature(key, payload):
+    """Compute the payload signature given a key."""
+    mac = hmac.new(key, msg=payload, digestmod=hashlib.sha1)
+    return mac.hexdigest()
+
+
+def signature_check(key, post_signature, payload):
+    """Check the HTTP POST legitimacy."""
+    if post_signature.startswith('sha1='):
+        sha_name, signature = post_signature.split('=')
+    else:
+        return False
+    if not signature:
+        return False
+    # HMAC requires its key to be bytes, but data is strings.
+    hexmac = get_payload_signature(key, payload)
+    return compare_digest(hexmac, signature.encode('utf-8'))
+
+
+def compare_digest(x, y):
+    """Create a hmac comparison.
+
+    Approximates hmac.compare_digest (Py 2.7.7+) until we upgrade.
+    TODO: SEE IF THIS NEEDS TO BE ADAPTED FOR PY3.6
```

Comment:
Maybe we can file a bug for this? Unless it's a leftover comment?

Reply:
@miketaylr This is weird. At first I thought it was a rebase snafu, but I opened the branch locally and it's not there. So I think it's just an earlier commit (there are 5 in here) of the same file? (Same for the secret key stuff below.)